fem/pgridfunc.cpp

```diff
@@ -225,19 +235,21 @@ void ParGridFunction::ExchangeFaceNbrData()
    MPI_Request *recv_requests = requests + num_face_nbrs;
    MPI_Status  *statuses = new MPI_Status[num_face_nbrs];
 
-   const double *h_data = this->HostRead();
-   for (int i = 0; i < send_data.Size(); i++)
+   auto d_data = this->Read();
```

Comment:
Would there be a way to use direct pointers when `GetGPUAwareMPI` is `true`?

Reply:
I added the `GPUAwareMPI` code.